File: solutions/HJK9810/2026-08-25/low/26931.java
import java.util.ArrayDeque;
import java.util.Scanner;

class Solution {
	public static void main(String args[]) throws Exception {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();

		for(int test_case = 1; test_case <= T; test_case++) {
      char[] line = sc.next().toCharArray();
      ArrayDeque<Character> stack = new ArrayDeque<>();

      for (char alpa : line) {
        stack.push(alpa);

        while (stack.size() > 1) {
          char last = stack.pop();
          if (stack.peek() == last) {
            stack.pop();
          } else {
            stack.push(last);
            break;
          }
        }
      }

      System.out.printf("#%d %d\n", test_case, stack.size());
		}
	}
}

File: solutions/HJK9810/2026-08-25/low/26931.py
def algorithm():
    line = list(input())
    stack = []

    for alpa in line:
        stack.append(alpa)
        if len(stack) < 2: continue

        while len(stack) > 1 and stack[-1] == stack[-2]:
            stack.pop()
            stack.pop()

    return len(stack)

T = int(input())
# 여러개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    print(f"#{test_case} {algorithm()}")
